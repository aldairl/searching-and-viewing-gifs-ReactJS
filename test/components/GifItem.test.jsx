const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components/GifItem");

describe("Test in GifItem", () => {
  const title = "title item";
  const url =
    "https://play-lh.googleusercontent.com/V_P-I-UENK93ahkQgOWel8X8yFxjhOOfMAZjxXrqp311Gm_RBtlDXHLQhwFZN8n4aIQ";
  test("should match the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("should be display the title", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });

  test("should display an image with url and alt especified", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug()
    // expect(screen.getByRole('img').src).toBe(url)

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
});
