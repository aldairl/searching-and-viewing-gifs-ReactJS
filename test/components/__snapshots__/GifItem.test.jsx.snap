// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title item"
      src="https://play-lh.googleusercontent.com/V_P-I-UENK93ahkQgOWel8X8yFxjhOOfMAZjxXrqp311Gm_RBtlDXHLQhwFZN8n4aIQ"
    />
    <p>
      title item
    </p>
  </div>
</div>
`;
